x-pack/plugins/ml/public/application/jobs/jobs_list/components/confirm_modals/start_datafeeds_confirm_modal.tsx

```diff
@@ -0,0 +1,142 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useState, useEffect, useCallback, useMemo } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiSpacer,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiText,
+} from '@elastic/eui';
+import { MlSummaryJob } from '../../../../../../common/types/anomaly_detection_jobs';
+import { isManagedJob } from '../../../jobs_utils';
+// @ts-ignore
+import { startDatafeeds } from '../utils';
+
+type ShowFunc = (jobs: MlSummaryJob[]) => void;
+
+interface Props {
+  setShowFunction(showFunc: ShowFunc): void;
+  unsetShowFunction(): void;
+  refreshJobs(): void;
+  showStartDatafeedModal(job: MlSummaryJob[]): void;
+}
+
+export const StartDatafeedsConfirmModal: FC<Props> = ({
+  setShowFunction,
+  unsetShowFunction,
+  refreshJobs,
+  showStartDatafeedModal,
+}) => {
+  const [modalVisible, setModalVisible] = useState(false);
+  const [isManaged, setHasManaged] = useState(true);
+  const [jobsToStart, setJobsToStart] = useState<MlSummaryJob[]>([]);
+
+  const jobIds = useMemo(() => jobsToStart.map(({ id }) => id), [jobsToStart]);
+
+  useEffect(() => {
+    if (typeof setShowFunction === 'function') {
+      setShowFunction(showModal);
+    }
+    return () => {
+      if (typeof unsetShowFunction === 'function') {
+        unsetShowFunction();
+      }
+    };
+  }, []);
+
+  const showModal = useCallback((jobs: MlSummaryJob[]) => {
+    setJobsToStart(jobs);
+
+    if (jobs.some((j) => isManagedJob(j))) {
+      setModalVisible(true);
+      setHasManaged(true);
+    }
+  }, []);
+
+  const closeModal = useCallback(() => {
+    setModalVisible(false);
+    setHasManaged(false);
+  }, []);
+
+  if (modalVisible === false) {
+    return null;
+  }
+
+  if (isManaged) {
+    const title = (
+      <FormattedMessage
+        id="xpack.ml.jobsList.startDatafeedsModal.startDatafeedsTitle"
+        defaultMessage="Start datafeed for {jobsCount, plural, one {{jobId}} other {# jobs}}?"
+        values={{
+          jobsCount: jobIds.length,
+          jobId: jobIds[0],
+        }}
+      />
+    );
+
+    return (
+      <EuiModal data-test-subj="mlStartDatafeedsConfirmModal" onClose={closeModal}>
+        <EuiModalHeader>
+          <EuiModalHeaderTitle>{title}</EuiModalHeaderTitle>
+        </EuiModalHeader>
+        <EuiModalBody>
+          <p>
+            <>
+              <EuiText>
+                <FormattedMessage
+                  id="xpack.ml.jobsList.startDatafeedsModal.startManagedDatafeedsDescription"
+                  defaultMessage="{jobsCount, plural, one {This job was} other {At least one of these jobs was}} deployed as part of a module; starting {jobsCount, plural, one {it} other {them}} might impact other parts of the product."
```

Comment:
> ... deployed as part of a module...

I am reconsidering whether someone seeing this would know what a "module" was. In the documentation, we refer to [(security) prebuilt jobs](https://www.elastic.co/guide/en/security/master/machine-learning.html#included-jobs) and [supplied configurations](https://www.elastic.co/guide/en/machine-learning/master/ootb-ml-jobs.html). In the ML app, they're labelled as "preconfigured jobs" in the job creation wizard. In the long run if this type of configuration is provided as part of integrations, that might be another term to consider here too.

Would it be clearer to say "This preconfigured job is provided by Elastic..." or else "This preconfigured job is provided in an Elastic integration..."?

Reply:
> Would it be clearer to say "This preconfigured job is provided by Elastic..." or else "This preconfigured job is provided in an Elastic integration..."?

++ "module" was confusing to me as I'd never heard the term before. If we externalize that word here, we'd need to update the docs as well. I like the idea of changing the text to one of Lisa's examples instead.

Reply:
++ to the use of `This preconfigured job is provided by Elastic..` here rather than 'module' which we only use internally.

Reply:
Updated here https://github.com/elastic/kibana/pull/122305/commits/427be5b2ff13c2453e9b03611547f978baa19cbc